utils/obsproc/Rads2Ioda.h

```diff
@@ -123,7 +128,30 @@ namespace gdasapp {
         (iodaVars.obsVal_ > -4.0 && iodaVars.obsVal_ < 4.0);
       iodaVars.trim(boundsCheck);
 
-      return iodaVars;
+      // Redating and adjusting Errors
+      char dash, colon, T, Z;
+      int jday, year, month, day, hour, minute, second;
+
+      std::istringstream dabegin(windowBegin_.toString());
+      dabegin >> year >> dash >> month >> dash >> day >> T
+         >> hour >> colon >> minute >> colon >> second >> Z;
+
+      uint64_t julianDate = util::datefunctions::dateToJulian(year, month, day);
+      int daysSinceEpoch = julianDate - 2400002;  // Since Epoch 1858-11-17
+      int secondsOffset = util::datefunctions::hmsToSeconds(hour, minute, second);
+
+      if (iodaVars.datetime_.size() == 0) {
+        oops::Log::info() << "datetime_ is empty" << std::endl;
+      } else {
+        // DA window begins
+        int64_t minDAwindow = (daysSinceEpoch*86400) + secondsOffset;
+        // DA window ends at hours of windowcyc later
+        int64_t maxDAwindow = minDAwindow + (windowcyc * 3600);
```

Comment:
compute the "maxdawindow" the same way you computed "mindawindow". Add a utility function somewhere to de-clutter your code.

Reply:
> compute the "maxdawindow" the same way you computed "mindawindow". Add a utility function somewhere to de-clutter your code.

Julian day conversion and sec is now placed in util.h